src/appfl/misc/utils.py

```diff
@@ -121,6 +123,55 @@ def save_model_iteration(t, model, cfg: DictConfig):
         uniq += 1
 
     torch.save(model, file)
+    
+
+def load_model_state(cfg: DictConfig, model, client_id = None):
+    
+    # This function allows to use partial model weights into a model.
+    # Useful since server model will only have shared layer weights when personalization is enabled.
+    if client_id == None:
+        file = cfg.load_model_dirname + "/%s%s" % (cfg.load_model_filename, ".pt")
+    else:
+        file = cfg.load_model_dirname + "/client_%d"%client_id + "/%s%s" % (cfg.load_model_filename, ".pt")
+
+    model.load_state_dict(torch.load(file),strict=False)
+        
+    return model
+    
+
+def save_model_state_iteration(t, model, cfg: DictConfig, client_id = None):
```

Comment:
Is this used for `personalized FL`? Can we combine this with `save_model_iteration`?

Reply:
Yes, it is used for `personalized FL`.

It will be a bit difficult to combine it with `save_model_iteration`. `save_model_iteration` currently saves the whole model, while `save_model_state_iteration` saves the `state_dict` of the model, and excludes the personalized weights from the `state_dict` if we are saving the server weights.

Reply:
One way to merge it with `save_model_iteration` would be to zero out the personalized weights in the server model before saving it.